### PR TITLE
Feature/add date format to parameters

### PR DESCRIPTION
**Pull request by SirJetboy**

### CODE REVIEW

File: lib/los_santos.rb [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
+  ALLOWED_CLASS = [String, Date, Time, DateTime].freeze
+
+  def self.name_day(date = Time.now)
     begin
-      date_split = date_string.split('-')
-      return data[date_split[1].to_i][date_split[0].to_i]
-    rescue NoMethodError
-      return print_usage
+      date = Date.strptime(date, '%d-%m') if date.class == String
+    rescue ArgumentError
```
Comment (twoneks) [inline lib/los_santos.rb]: I think you can remove this begin rescue

File: lib/los_santos.rb [hunk 2]
```diff
@@ -2,19 +2,18 @@
...
+      raise ArgumentError, "String usage: LosSantos.name_day 'dd-mm'"
     end
-  end
-
-  private
-
-  def self.print_usage
-    puts 'Usage: LosSantos.name_day [\'dd-mm\']'
+    unless ALLOWED_CLASS.include? date.class
+      raise ArgumentError, 'Only String, Date, Time and DateTime are allowed'
```
Comment (twoneks) [inline lib/los_santos.rb]: Use `ALLOWED_CLASS ` variable to make this string dynamic:
`raise ArgumentError, "Only #{ALLOWED_CLASS.join(' ,')} are allowed"`

**Review by twoneks — CHANGES_REQUESTED** [review]

File: lib/los_santos.rb (hunk 1 above)
Comment (SirJetboy) [reply lib/los_santos.rb]: I put this begin rescue to control the error and display a custom error message

File: lib/los_santos.rb (hunk 2 above)
Comment (SirJetboy) [reply lib/los_santos.rb]: :ok:

Comment (ryanfox1985) [reply lib/los_santos.rb]: This check should be the first check.

File: lib/los_santos.rb (hunk 1 above)
Comment (ryanfox1985) [reply lib/los_santos.rb]: I should accept strings like => `2011-05-19` you have quickly conversions like => 

```
date = Date.parse("2011-05-19 10:30:14")
date = Date.parse("2011-05-19")
```

File: lib/los_santos.rb [hunk 3]
```diff
@@ -2,19 +2,18 @@
...
+    end
+    data = YAML.load_file('lib/los_santos/data/fr.yml')
+    data[date.month][date.day]
```
Comment (ryanfox1985) [inline lib/los_santos.rb]: be carefully if you don't have a month present this raises and nil exception.

File: lib/los_santos.rb (hunk 1 above)
Comment (SirJetboy) [reply lib/los_santos.rb]: I thought to use parse at the beginning, but I don't like its behavior:  
Date.parse('19-05')
Date.parse('05-19')
Date.parse('19-05-2017')
Date.parse('2017-05-19')
=> three different dates
If you don't precise the year, the date is gonna be bad, and we don't need the year in my gem. It was tricky to manage with parse so I just allow one format: 'dd-mm'

File: lib/los_santos.rb (hunk 3 above)
Comment (SirJetboy) [reply lib/los_santos.rb]: Is it possible to have a Date, Time or DateTime without a month?

File: lib/los_santos.rb (hunk 2 above)
Comment (SirJetboy) [reply lib/los_santos.rb]: Yep :ok_hand:

File: lib/los_santos.rb [hunk 4]
```diff
@@ -2,19 +2,18 @@
...
-  end
-
-  private
-
-  def self.print_usage
-    puts 'Usage: LosSantos.name_day [\'dd-mm\']'
+    unless ALLOWED_CLASS.include? date.class
+      raise ArgumentError, 'Only String, Date, Time and DateTime are allowed'
+    end
+    data = YAML.load_file('lib/los_santos/data/fr.yml')
```
Comment (albertlb) [inline lib/los_santos.rb]: It's better to have this loaded in another method so it can be reused

File: lib/los_santos.rb [hunk 5]
```diff
@@ -2,19 +2,18 @@
 require 'yaml'
 
 module LosSantos
-  def self.name_day(date_string = Time.now.strftime('%d-%m'))
-    data = YAML.load_file('lib/los_santos/data/fr.yml')
+  ALLOWED_CLASS = [String, Date, Time, DateTime].freeze
+
+  def self.name_day(date = Time.now)
```
Comment (albertlb) [inline lib/los_santos.rb]: it would be nice to have a method "name_day?" that returns you whether it's you day name or not

Comment (albertlb) [reply lib/los_santos.rb]: Also a control over the locale

File: lib/los_santos.rb (hunk 3 above)
Comment (ryanfox1985) [reply lib/los_santos.rb]: data[date.month] can be nil.

**Review by albertlb — APPROVED** [review]
nice!

**Pull request merged**